File: backend/src/main/java/hu/ibc/ibcpartners/partner/dto/CommissionDto.java
package hu.ibc.ibcpartners.partner.dto;

import hu.ibc.ibcpartners.partner.entity.CommissionStatus;

import java.time.LocalDate;

public record CommissionDto(
        Long id,
        Long transactionId,
        Long userId,
        String userName,
        String sellerName,
        String description,
        LocalDate fulfillmentDate,
        Long amount,
        Long commission,
        CommissionStatus status) {
}

File: frontend/src/app/features/dashboard/user/profile/commissions/commission-dto.ts
import {CommissionStatus} from '../../../../../shared/commission-status';

export interface CommissionDto {
  id: number;
  transactionId: number;
  userId: number;
  userName: string;
  sellerName: string;
  description: string | null;
  fulfillmentDate: string | null;
  amount: number;
  commission: number;
  status: CommissionStatus;
}
